src/java/map/BinaryMask.java

```diff
@@ -93,6 +96,41 @@ public void set(int x, int y, boolean value) {
         mask[x][y] = value;
     }
 
+    public float getDistanceBetweenPoints(int x1, int y1, int x2, int y2) {
+        return (float) StrictMath.sqrt((double) ((x1 - x2) ^ 2 + (y1 - y2) ^ 2));
+    }
+
+    public float getDistanceBetweenPoints(int x1, int y1, Point Point2) {
+        return (float) StrictMath.sqrt((double) ((x1 - Point2.x) ^ 2 + (y1 - Point2.y) ^ 2));
+    }
+
+    public float getDistanceBetweenPoints(Point Point1, Point Point2) {
+        return (float) StrictMath.sqrt((double) ((Point1.x - Point2.x) ^ 2 + (Point2.x - Point2.y) ^ 2));
+    }
+
+    public Point getNearestPointInAreaTo(int x, int y) {
+        Point nearestPoint = new Point (x, y);
+        int rank;
+        FloatMask distanceField = getDistanceFieldFromPointInArea(x, y);
+        for (int a = 0; a < getSize(); a++) {
+            for (int b = 0; b < getSize(); b++) {
+                if (this.get(a, b)) {
+                    rank = distanceField.getRankOfPointInArea(a, b, this);
+                    if(rank == 1) {
+                        nearestPoint.setLocation(a, b);
+                        a = getSize();
+                        b = getSize();
+                    }
+                }
+            }
+        }
+        return nearestPoint;
+    }
+
+    public float getLowestDistanceFromPointToArea(int x, int y) {
```

Comment:
This is the same as getDistanceField().get(x,y)

Reply:
function has been changed - I could do getDistanceField().get(x, y) instead if you think that would be more efficient than what I have now?

Reply:
this will be fine and more efficient than that I think